src/realtime/price.updates.ts

```diff
@@ -209,35 +198,40 @@ export class PriceUpdates {
         const that = this;
 
         setTimeout(() => {
-            if (!isEmpty(that.subscribersWithTicker)) {
-                that.subscribersWithTicker.forEach((symbolTicker) => {
-                    log(`cancelMktData ${symbolTicker.symbol} tickerId=${symbolTicker.tickerId}`);
-                    that.ib.cancelMktData(symbolTicker.tickerId);
-                });
+            for (const symbolTicker of Object.values(this.tickerIdToData)) {
+                log(`cancelMktData ${symbolTicker.symbol} tickerId=${symbolTicker.tickerId}`);
+                that.ib.cancelMktData(symbolTicker.tickerId);
             }
-            return;
+            this.tickerIdToData = {};
+            this.keyToTickerId = {};
         }, 1000);
     }
 
     /**
      * unsubscribe
      */
-    public unsubscribe(symbol: string): void {
+    public unsubscribe(contract: {readonly conId?: number; readonly symbol?: string}): void {
         const that = this;
 
-        if (isEmpty(symbol)) {
+        const {conId, symbol} = contract;
+        const key = conId ?? symbol;
+        if (!key) {
+            return;
+        }
+
+        const tickerId = this.keyToTickerId[key];
+        if (!tickerId) {
             return;
         }
 
         setTimeout(() => {
-            if (!isEmpty(that.subscribersWithTicker)) {
-                const symbolTicker = that.subscribersWithTicker.find((st) => st.symbol === symbol);
-                if (symbolTicker) {
-                    log(`cancelMktData ${symbolTicker.symbol} tickerId=${symbolTicker.tickerId}`);
-                    that.ib.cancelMktData(symbolTicker.tickerId);
-                }
+            const symbolTicker = this.tickerIdToData[tickerId];
+            if (symbolTicker) {
+                log(`cancelMktData ${symbolTicker.symbol} tickerId=${symbolTicker.tickerId}`);
+                that.ib.cancelMktData(symbolTicker.tickerId);
             }
-            return;
+            delete this.tickerIdToData[tickerId];
```

Comment:
I would avoid using `this` in here,

Reply:
Looking over it again, the use of `this` is in a lambda function, which is safe by my understanding, and I haven't had to use the `that` workaround in lambda functions in other projects.  I'd rather not use the `that` workaround if it doesn't actually make a difference...  What do you think?